addons/web/static/src/model/relational_model/record.js

```diff
@@ -412,28 +410,37 @@ export class Record extends DataPoint {
                         this._isRequired(fieldName) &&
                         (!this.data[fieldName] || !Object.keys(this.data[fieldName]).length)
                     ) {
-                        unsetRequiredFields.push(fieldName);
+                        unsetRequiredFields.add(fieldName);
                     }
                     break;
                 }
                 default:
                     if (!this.data[fieldName] && this._isRequired(fieldName)) {
-                        unsetRequiredFields.push(fieldName);
+                        unsetRequiredFields.add(fieldName);
                     }
             }
         }
 
         if (silent) {
-            return !unsetRequiredFields.length;
+            return !unsetRequiredFields.size;
         }
 
-        for (const fieldName of Array.from(this._unsetRequiredFields)) {
-            this._invalidFields.delete(fieldName);
-        }
-        this._unsetRequiredFields.clear();
-        for (const fieldName of unsetRequiredFields) {
-            this._unsetRequiredFields.add(fieldName);
-            this._setInvalidField(fieldName);
+        if (removeInvalidOnly) {
+            for (const fieldName of Array.from(this._unsetRequiredFields)) {
```

Comment:
why not use this._invalidFields ?

Reply:
field invalid (bad type value)

Reply:
because we don't want to mark valid a field that is set, but to an invalid valid